babel.config.js

```diff
@@ -47,14 +50,25 @@ module.exports = (api) => {
   // Options for the module-resolver plugin.
   // Used for resolving source files during development.
   const resolverOptions = {
-    alias: {
-      ...(DEV || TEST
-        ? {
+    ...(DEV || TEST
+      ? {
+          alias: {
             '^@apache-annotator/([^/]+)$': ([, name]) =>
-              path.join(__dirname, 'packages', name, '/src/index.ts'),
-          }
-        : null),
-    },
+              path.join(packagePath, name, '/src/index.ts'),
+          },
+          resolvePath(sourcePath, currentFile, opts) {
+            if (
+              currentFile.startsWith(packagePath) &&
+              currentFile.endsWith('.ts') &&
+              sourcePath.startsWith('.') &&
+              sourcePath.endsWith('.js')
+            ) {
+              return sourcePath.replace(/\.js$/, '.ts');
+            }
+            return resolvePath(sourcePath, currentFile, opts);
+          },
```

Comment:
Unfortunate we’d need a custom resolver for this, but so be it. I wonder why TypeScript does not simply accept an `import '….ts';` statement, and would transpile it to the equivalent `.js` import.

Reply:
I don't recall the arguments, but I have read through the issues about it. It doesn't seem likely. They don't want to get into the business of rewriting imports. Their position is that you should write `.js`. The TypeScript resolver knows how to resolve that to equivalent `.ts` files, and maybe part of the answer is to use `eslint-import-resolver-typescript`. However, it's helpful to have Babel for our final build anyway, and `babel-plugin-module-resolver` _will_ rewrite imports for us, and `eslint-import-resolver-babel-module` can piggyback off that, so this is what I think is the best thing for now.